crates/core/src/key/common.rs

```diff
@@ -43,6 +54,18 @@ pub enum PublicKey {
     Secp256k1(secp256k1::PublicKey),
 }
 
+impl BorshSchema for PublicKey {
+    fn add_definitions_recursively(
+        definitions: &mut BTreeMap<Declaration, Definition>,
+    ) {
+        CommonPublicKey::add_definitions_recursively(definitions)
+    }
+
+    fn declaration() -> Declaration {
+        CommonPublicKey::declaration()
+    }
+}
```

Comment:
if we go with the second approach, we can simply derive `BorshSchema` and remove this `impl` block, I think